src/tracing/esp32_trace/counter.h

```diff
@@ -0,0 +1,58 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <esp_diagnostics_metrics.h>
+#include <esp_log.h>
+#include <string.h>
+
+#define PATH1 "sys.cnt"
+namespace Insights {
+class ESPInsightsCounter
+{
+private:
+    static ESPInsightsCounter * mHead;
+    char label[50];
```

Comment:
do these need to be static sized? trace macros are generally passed const char *, so these could retain the pointer instead (or do we expect dynamic names/groups?)

Reply:
We expect dynamic names for label and group to be assigned based on the trace macro.

Reply:
Please provide an example. otherwise please apply `You ain't gonna need it` - if we do not need it now, please do not implement "for the future" because most often the future never needs it. If we do need it, switching from `const char *` to alocated memory and string copy should be easy.

Reply:
I have addressed the comment in the latest commit.